subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/GeneratePluginDescriptors.java

```diff
@@ -58,11 +58,20 @@ public GeneratePluginDescriptors() {
         );
     }
 
+    /**
+     * Because of {@link PluginDeclaration} is not cacheable, the caching is handled by {@link #getImplementationClassById()}
```

Comment:
This is not about `PluginDeclaration` not being cacheable. I don't know how a class can be "cacheable". And I didn't realize that `getDeclarations()` doesn't have any Javadoc :(.

So what I'd add here is that what the property means for the user. Something like:

```
The PluginDeclarations for which to create the descriptors.

The task will create a descriptor for each of the PluginDeclaration added here.
```

I think the information about the inputs is better captured in the comment in the `@Internal` annotation. Maybe that would need some improvement?
```

Reply:
Now having a second read-through, I agree that this comment is a little bit overkill, but maybe instead of this comment.
The JavaDoc suggestion is nice, but if I would go with that one, I would change the `@Internal` annotation's message to:
> Changes for the declarations are tracked via implementationClassById

As my gripe with this whole situation is that it wasn't obvious what's happening with the caching in this class.

Reply:
Sounds good.